nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/scroll/ScrollDescriptor.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     bdelbosc
+ */
+
+package org.nuxeo.ecm.core.scroll;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.nuxeo.common.xmap.annotation.XNode;
+import org.nuxeo.common.xmap.annotation.XNodeMap;
+import org.nuxeo.common.xmap.annotation.XObject;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.api.scroll.Scroll;
+import org.nuxeo.runtime.model.Descriptor;
+
+/**
+ * @since 11.1
+ */
+@XObject("scroll")
+public class ScrollDescriptor implements Descriptor {
+
+    @XNode("@enabled")
+    protected boolean isEnabled = true;
+
+    @XNode("@name")
+    protected String name;
+
+    @XNode("@default")
+    protected boolean isDefault;
+
+    @XNode("@class")
+    protected Class<? extends Scroll> scrollClass;
+
+    @XNodeMap(value = "option", key = "@name", type = HashMap.class, componentType = String.class)
+    protected Map<String, String> options = new HashMap<>();
+
+    protected Map<String, String> optionsReadOnly;
```

Comment:
This seems unused.

Reply:
this is the readonly version of the option map that is returned on getOptions()

Reply:
Why is it worth storing it? You can return `Collections.unmodifiableMap(options)` everytime. This seems like a premature optimization.